### PR TITLE
feat: add support for custom audience in aws oidc provider

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds support for custom audience in aws oidc provider.

Currently, the audience is set to `amazonaws.com` and cannot be changed.
It can be useful to set it to a custom value.
For example, by default, the EKS pod identity webhook uses `sts.amazonaws.com`.

### CODE REVIEW

File: pkg/apis/kops/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,4 +1,3 @@
-//go:build !ignore_autogenerated
```
Comment (eddycharly) [inline pkg/apis/kops/zz_generated.deepcopy.go:1]: Don't know why `make apimachinery` removed those lines.

Comment (johngmyers) [reply pkg/apis/kops/zz_generated.deepcopy.go]: You need to use Go 1.17.1

Comment (eddycharly) [reply pkg/apis/kops/zz_generated.deepcopy.go]: Thanks, fixed now.

**Comment by eddycharly** [comment]
@olemarkus i think it will break addons using irsa

**Comment by olemarkus** [comment]
You could append additional audiences rather than replace.
or validation can check that is irsa is enabled, the audience list contains amazonaws

**Comment by eddycharly** [comment]
I went with the additional audiences approach.
Now i need to update the AWS task to take care of changes.

**Comment by eddycharly** [comment]
`pull-kops-e2e-aws-ebs-csi-driver-irsa` is failing but i'm not sure it is related to this PR.

**Comment by olemarkus** [comment]
yeah sorry, it was a bad example. There is a lot of flaking tests in that one.

**Comment by rifelpet** [comment]
Could we add this to an integration test?

Add example values in both `in-*.yaml` files [here](https://github.com/kubernetes/kops/tree/master/tests/integration/update_cluster/complex) and rerun ./hack/update-expected.sh. The resulting terraform and cloudformation files in the same directory should get updated.

**Comment by eddycharly** [comment]
Thanks @rifelpet it looks like i didn't update everything as the terraform files didn't change as expected.

**Comment by eddycharly** [comment]
@rifelpet the test you mentioned didn't have irsa enabled, i added the test [here](https://github.com/kubernetes/kops/tree/master/tests/integration/update_cluster/irsa) instead.

**Comment by eddycharly** [comment]
Ah, macos build is failing 🤔 
Any way to restart it ?

**Comment by eddycharly** [comment]
I squashed the commits, it should be ready for review.

File: pkg/apis/kops/cluster.go [hunk 2]
```diff
@@ -223,6 +223,8 @@ type ServiceAccountIssuerDiscoveryConfig struct {
 	DiscoveryStore string `json:"discoveryStore,omitempty"`
 	// EnableAWSOIDCProvider will provision an AWS OIDC provider that trusts the ServiceAccount Issuer
 	EnableAWSOIDCProvider bool `json:"enableAWSOIDCProvider,omitempty"`
+	// AdditionalAudiences adds user defined audiences to the provisionned AWS OIDC provider
```
Comment (rifelpet) [inline pkg/apis/kops/cluster.go:226]:
```suggestion
	// AdditionalAudiences adds user defined audiences to the provisioned AWS OIDC provider
```

File: k8s/crds/kops.k8s.io_clusters.yaml [hunk 3]
```diff
@@ -4332,6 +4332,12 @@ spec:
                 description: ServiceAccountIssuerDiscovery configures the OIDC Issuer
                   for ServiceAccounts.
                 properties:
+                  additionalAudiences:
+                    description: AdditionalAudiences adds user defined audiences to
+                      the provisionned AWS OIDC provider
```
Comment (rifelpet) [inline k8s/crds/kops.k8s.io_clusters.yaml:4337]:
```suggestion
                      the provisioned AWS OIDC provider
```

File: pkg/apis/kops/v1alpha2/cluster.go [hunk 4]
```diff
@@ -221,6 +221,8 @@ type ServiceAccountIssuerDiscoveryConfig struct {
 	DiscoveryStore string `json:"discoveryStore,omitempty"`
 	// EnableAWSOIDCProvider will provision an AWS OIDC provider that trusts the ServiceAccount Issuer
 	EnableAWSOIDCProvider bool `json:"enableAWSOIDCProvider,omitempty"`
+	// AdditionalAudiences adds user defined audiences to the provisionned AWS OIDC provider
```
Comment (rifelpet) [inline pkg/apis/kops/v1alpha2/cluster.go:224]:
```suggestion
	// AdditionalAudiences adds user defined audiences to the provisioned AWS OIDC provider
```

**Review by olemarkus — APPROVED** [review]

**Pull request merged**